### ISSUE TITLE
Example in wiki under SC2046 causes warnings

### ISSUE THREAD

**Issue by scallopedllama** [koalaman/shellcheck#1888]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2046
- My shellcheck version (`shellcheck --version` or "online"): n/a
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

This example is at the end of the wiki page for [SC2046](https://github.com/koalaman/shellcheck/wiki/SC2046):
```
compile () {
    args=( $(pkg-config --libs openssl) "${@}" )
    gcc "${args[@]}" client.c
}
compile -DDEBUG
+ gcc -lssl -lcrypto -DDEBUG client.c
```

#### Here's what shellcheck currently says:

```
In test2.sh line 3:
    args=( $(pkg-config --libs openssl) "${@}" )
           ^--------------------------^ SC2207: Prefer mapfile or read -a to split command output (or quote to avoid splitting).

For more information:
  https://www.shellcheck.net/wiki/SC2207 -- Prefer mapfile or read -a to spli...
```

#### Here's what I wanted or expected to see:

The wiki page implies that this will be fine but generates the warning above

**Comment by matthewpersico** [comment]
Huh? I only see one place on that page where the wiki page *might* be implying that:

> If the output should be a single element:
> 
> array=( "$(mycommand)" )

And that does not apply to your command because you have not quoted it like `args=( "$(pkg-config --libs openssl)" "${@}" )`.

**Comment by scallopedllama** [comment]
Sorry for not being more clear. The code block I posted isn't my code, it's copied from the wiki page for SC2046.

Taking the last example on the wiki page verbatim and putting it into shellcheck (including the online version):

```shell
#!/bin/bash
compile () {
    args=( $(pkg-config --libs openssl) "${@}" )
    gcc "${args[@]}" client.c
}
compile -DDEBUG
+ gcc -lssl -lcrypto -DDEBUG client.c
```

Generates the warning I posted. Seeing this block as an example on the wiki page made me think that it would not generate a warning but it does, it's a little misleading.

**Comment by scallopedllama** [comment]
To be clear: I just think the wiki example should be updated. I don't believe there is an issue with shellcheck itself.

**Comment by brother** [comment]
It's a wiki - next time just edit it. I've done it this time =)

https://github.com/koalaman/shellcheck/wiki/SC2046

**Comment by scallopedllama** [comment]
Will do
